utils/src/main/java/dev/arkbuilders/components/utils/PathExt.kt

```diff
@@ -1,10 +1,17 @@
 package dev.arkbuilders.components.utils
 
 import java.nio.file.Path
+import kotlin.io.path.Path
+
+val INTERNAL_STORAGE = Path("/storage/emulated/0")
```

Comment:
Firstly, this path actually denotes `external storage`, not `internal storage`, so you might want to update the value and the function's name. Also, the intention is to check if a path is the `root` external storage (not every external storage path is a root one). 
So, the name `isInternalStorage` is quite confusing.
What about this name? `isRootExternalStorage`?

Secondly, comparison using this hard-coded path is unreliable because the path is not always exactly the same on every device.
Instead, what do you think about the other approach as below?

```
fun File.isRootExternalStorageFolder(): Boolean {
    // Root of external storage
    val rootExternalStorage = Environment.getExternalStoragePublicDirectory(Environment.DIRECTORY_DOWNLOADS).parentFile
    return this.canonicalPath == rootExternalStorage?.canonicalPath
}
```
I guess this will work on every device regardless of the actually path value of the external storage.

Reply:
Internal storage is meant in relation to sd card, for example there is device and sd card, the device will be internal.
Naming taken from [there](https://github.com/TeamAmaze/AmazeFileManager/blob/5cb924a11fb72b2f7ac9eaee3ddce5c62795a9bb/file_operations/src/main/java/com/amaze/filemanager/fileoperations/filesystem/StorageNaming.java#L47)

With arklib we can get list of storage paths (device, sd card)
https://github.com/ARK-Builders/arklib-android/blob/main/lib/src/main/java/dev/arkbuilders/arklib/utils/FileUtils.kt
`/storage/emulated/0`  is needed only to show meaningful `Internal storage` instead of full path. User will not be able to go to `/storage/emulated/`, `/storage/`, `/`
Hardcode is bad, but it was simple option that covers 95% of cases
User doesn't know about internal folder of Android apps :)

How can I understand with your code whether this path belongs to device or sd card?

Actually `INTERNAL_STORAGE` constant is not needed here, it is already in arklib.
And I'll delete `isInternalStorage`, it is really confusing.

I did some research, there is an api that seems to let find out storage name:
https://developer.android.com/reference/android/os/storage/StorageVolume
https://github.com/zhanghai/MaterialFiles/blob/master/app/src/main/java/me/zhanghai/android/files/storage/DeviceStorage.kt

We can create a task to refactor this constant and get names through normal api.
@kirillt  what do you think?